browser/brave_shields/brave_shields_web_contents_observer.cc

```diff
@@ -317,25 +282,41 @@ void BraveShieldsWebContentsObserver::SetReceiverImplForTesting(
   g_receiver_impl_for_testing = impl;
 }
 
+void BraveShieldsWebContentsObserver::SendShieldsSettings(
+    NavigationHandle* navigation_handle) {
+  DCHECK(navigation_handle);
+  RenderFrameHost* rfh = navigation_handle->GetRenderFrameHost();
+
+  const GURL& primary_url =
+      navigation_handle->GetParentFrameOrOuterDocument()
+          ? navigation_handle->GetParentFrameOrOuterDocument()
+                ->GetOutermostMainFrame()
+                ->GetLastCommittedURL()
+          : navigation_handle->GetURL();
+
+  const brave_shields::mojom::FarblingLevel farbling_level =
+      brave_shields::GetFarblingLevel(
+          HostContentSettingsMapFactory::GetForProfile(
+              rfh->GetBrowserContext()),
+          primary_url);
+
+  PrefService* pref_service =
+      user_prefs::UserPrefs::Get(rfh->GetBrowserContext());
+
+  mojo::AssociatedRemote<brave_shields::mojom::BraveShields> agent;
+  rfh->GetRemoteAssociatedInterfaces()->GetInterface(&agent);
+  agent->SetShieldsSettings(brave_shields::mojom::ShieldsSettings::New(
+      farbling_level, allowed_scripts_,
+      brave_shields::IsReduceLanguageEnabledForProfile(pref_service)));
```

Comment:
IsReduceLanguageEnabledForProfile seems like it belongs in BraveRendererUpdater ?

Reply:
data passed by `BraveRendererUpdater` is not accessible by blink internals.

Reply:
not directly, but it could be exposed through our own class just like we have done in other places